src/backends/redis.ts

```diff
@@ -45,10 +45,14 @@ export default class RedisBackend implements CeleryBackend {
    * @param {object} opts the options object for redis connect of ioredis
    */
   constructor(url: string, opts: object) {
-    this.redis = new Redis({
-      ...redisOptsFromUrl(url),
-      ...opts
-    });
+    if (url.startsWith("rediss://")){
+      this.redis = new Redis(url, {...opts});
+    } else {
+      this.redis = new Redis({
+        ...redisOptsFromUrl(url),
+        ...opts
+      });
+    }
   }
```

Comment:
Is there some reason not to use redisOptsFromUrl if it startsWith "rediss://" ?

Reply:
Thanks for reviewing!

If we use `redisOptsFromUrl` then the underlying `ioredis` is not aware that it should be a TLS connection (the protocol/url are not passed with the returned options). We would have to pass additional `tls:{}` options.

It might also be worth noting that, as far as I can tell, the `redisOptsFromUrl` is not needed for `redis://` urls either. Simply using `this.redis = new Redis(url, {...opts});` for `redis://` and `rediss://` works as expected (I have tested this, including simply passing `redis://` as the url).
Under the hood `ioredis` essentially performs the same parsing logic on urls as `redisOptsFromUrl` .

Reply:
On a separate note, I see there is an unrelated linting error in the build jobs. Would you like me to fix that in this MR?

Reply:
> It might also be worth noting that, as far as I can tell, the redisOptsFromUrl is not needed for redis:// urls either. Simply using this.redis = new Redis(url, {...opts}); for redis:// and rediss:// works as expected (I have tested this, including simply passing redis:// as the url).
Under the hood ioredis essentially performs the same parsing logic on urls as redisOptsFromUrl

Without `redisOptsFromUrl`, as I remember, `ioredis` do not recognize port or db if there's only url info so we should pass it to opts via `redisOptsFromUrl` but it seems ok now. I'll update and test it. Thank you.

Reply:
> On a separate note, I see there is an unrelated linting error in the build jobs. Would you like me to fix that in this MR?

It would be grateful if you did so :)

Reply:
> Without redisOptsFromUrl, as I remember, ioredis did not recognize port or db if there's only url info so we should pass it to opts via redisOptsFromUrl but it seems ok now

I understand, that makes sense! 
I have only tested with `ioredis` version `4.14.1`, which is the version currently used.

Looking through the `ioredis` changelog, I wonder if it makes sense to update the version to `4.28.x`. This introduces redis 6 compatibility and, in particular, the ability to include the username in the url (if also adding `?allowUsernameInURI=true`) which is useful with redis 6 ACLs. Perhaps this is something for a separate PR?

Reply:
Ok. I suppose it should be separated.
Thank you for your detailed explanation. It really helps for understanding :)

Reply:
@actumn i am very happy to work on other PRs for updating ‘ioredis’ version and/or removing ‘tedisOotsFromUrl’ usage. 
Please let me know if you would like me to. 
Again, many thanks for such a useful package!